lib/comments.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+exports.convertNodeComments = convertNodeComments;
+exports.convertProgramComments = convertProgramComments;
+
+function convertNodeComments(node) {
+    const {comments} = node;
+    
+    if (!comments)
+        return;
+    
+    delete node.comments;
+    node.leadingComments = undefined;
+    node.trailingComments = undefined;
+    node.innerComments = undefined;
+    
+    for (const comment of comments) {
+        // Default to leading comment.
+        const group = getCommentGroup(comment);
+        
+        if (!node[group])
+            node[group] = [];
+        
+        delete comment.leading;
+        delete comment.trailing;
+        comment.type = getCommentType(comment);
+        node[group].push(comment);
+    }
+}
+
+function convertProgramComments(comments) {
+    return comments.map((comment) => {
+        comment.type = getCommentType(comment);
+        return comment;
+    });
+}
+
+function getCommentType({type}) {
+    if (type === 'Line')
+        return 'CommentLine';
+    
+    if (type === 'Block')
+        return 'CommentBlock';
+    
+    // Assume an unknown type, or that it’s already a Babel type.
+    return type;
```

Comment:
```suggestion
```

I think we can remove this line if it can't be covered, it always can be improved in future, if there will be a need for this.